synapse/handlers/room_member.py

```diff
@@ -1634,7 +1677,13 @@ async def _make_and_store_3pid_invite(
         )
         return event, stream_id
 
-    async def _is_host_in_room(self, current_state_ids: StateMap[str]) -> bool:
+    async def _is_host_in_room(
+        self, is_partial_state_room: bool, current_state_ids: StateMap[str]
+    ) -> bool:
+        # When we only have a partial state, let's assume we are still in the room
+        if is_partial_state_room:
+            return True
+
```

Comment:
I think this means that #12802 is going to have to ensure this returns False as soon as the last person has left? Feels like there's race potential

Reply:
I think(?) you've addressed this by making it the caller's responsibility to interpret the return value of this function.